### PR TITLE
Enable GHES support, general clean-up.

### PR DESCRIPTION
**Pull request by ccstolley**
@bmaca pointed out in #21 that there is no ability to use GHES endpoints.
He also proposed a solution where the git remote hostname could be
used to determine if the API is github.com or not.
This PR implements his proposal, while also cleaning up a few
oddities in the code that I found while I was in there.

### CODE REVIEW

File: ghub.py [hunk 1]
```diff
@@ -168,77 +172,65 @@ def __fallback_get_api_token():
...
-        _, user_name_repo = repo_line.split(':')
-        while user_name_repo.startswith('/'):
-            user_name_repo = user_name_repo[1:]
-        user_name, repo = user_name_repo.split('/')[-2:]
-        repo = repo.replace('.git', '')
-    return user_name, repo
+        githost, path = url.split(':', maxsplit=1)
+        host = githost.split('@', maxsplit=1)[-1]
+        username, repo = path.split('/', maxsplit=1)
+        if repo.endswith('.git'): repo = repo.replace('.git', '')
```
Comment (bmaca) [inline ghub.py:208]: I think that one liner is anti pattern in python. 
Probably best to 
```
if repo.endswith('.git'): 
   repo = repo.replace('.git', '')
```
or the pythonic one liner is `repo = repo.replace('.git', '') if repo.endswith('.git') else repo`

**Review by bmaca — APPROVED** [review]
@ccstolley PR looks good to me. Small comment more tylic issue than anything. Thanks for looking into this.

File: test_ghub.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
 import os
 import unittest
 from io import StringIO
-from mock import patch
+from unittest.mock import patch
```
Comment (bmaca) [inline test_ghub.py:4]: With this you probably don't need `test_requirements.txt` might need to also update the build plan.

File: ghub.py (hunk 1 above)
Comment (ccstolley) [reply ghub.py:208]: Yeah, also `repo.replace('.git', '')` could be tightened to just `repo = repo[:-4]`. I'll fix both.

File: test_ghub.py (hunk 2 above)
Comment (ccstolley) [reply test_ghub.py:4]: Fixed, thanks for catching that.

**Pull request merged**